doc/specs/#1620 - Installer selector on multiple matches.md

```diff
@@ -0,0 +1,270 @@
+---
+author: Darshan Rander @SirusCodes
+created on: 2022-10-29
+last updated: 2022-11-11
+issue id: 1620
+---
+
+# Spec Title
+
+For [#1620](https://github.com/microsoft/winget-cli/issues/1620)
+
+## Abstract
+
+Currently, the winget.exe client simply lists all the matches and if the user wants to install the app they need to re-run the command with Id.
+
+The UX can be improved here by giving users an option to select the package instead of asking them to re-run the command with Id.
+
+This can be extended to `upgrade`, `show` and `uninstall` commands.
+
+## Inspiration
+
+Whenever I use Winget to install any application there are high chances that there are name collisions with other names and I get multiple suggestions in the table and it's annoying to copy the ID and rerun the command. I guess this is the case for a lot of others as well.
+
+## Solution Design
+
+If we find multiple packages then the user should get a selector.
+
+> It should not be shown if `--diable-interactivity` is set.
+
+### Settings
+
+The user needs to **opt-in** to this feature.
+
+Winget can use `--prompt` and `--no-prompt` flags to enable and disable it respectively.
+
+The user can change the default setting to always show the prompt by -
+
+```pwsh
+"packageDisambiguation": "prompt"
+```
+
+and disable it by -
+
+```pwsh
+"packageDisambiguation": "no-prompt"
+```
+
+### Valid inputs
+
+- Number - It should match with a _row number_ (0 will exit the selector)
+- Text - It should match either the package _Name_ or _Id_
+
+#### Flags
+
+- `--more` or `-m` flag - It should be used to show more n number of results.
+- `--id` - Filter results by id
+- `--name` - Filter results by name
+- `--moniker` - Filter results by the moniker
+- `--tag` - Filter results by tag
+- `--cmd` or `--command` - Filter results by command
+- `-s` or `--source` - Find the package using the specified source
+- `-e` or `--exact` - Find the package using exact match
+
+### Invalid inputs
+
+If a user enters an invalid value it should show an error asking the user for input again,
```

Comment:
I think this could be a point of contention. If using the value as additional refinement, I would prefer it to exit with the message `No package found matching input criteria`. However, this would be a UX choice that denelon or someone else at MSFT would need to make

Reply:
@denelon or @yao-msft can you please suggest something here?

Reply:
Are you looking for a suggestion on the error or something related to the heuristic for determining error cases?

Reply:
@denelon - Already resolved it with my recent PR into the forked branch